experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py

```diff
@@ -2957,6 +2957,104 @@ def test_desktop_large_prefs(
+            feature_configs=[
+                NimbusFeatureConfigFactory.create_desktop_prefflips_feature()
+            ],
+            channel=NimbusExperiment.Channel.NO_CHANNEL,
+        )
+
+        serializer = NimbusReviewSerializer(
+            experiment,
+            data=NimbusReviewSerializer(experiment, context={"user": self.user}).data,
+            context={"user": self.user},
+        )
+
+        self.assertFalse(serializer.is_valid())
+        self.assertEqual(
+            serializer.errors,
+            {"channel": [NimbusExperiment.ERROR_DESKTOP_PREFFLIPS_CHANNEL_REQUIRED]},
+        )
+
+    @parameterized.expand(
+        (
+            NimbusExperiment.Channel.UNBRANDED,
+            NimbusExperiment.Channel.NIGHTLY,
+            NimbusExperiment.Channel.BETA,
+            NimbusExperiment.Channel.RELEASE,
+            NimbusExperiment.Channel.ESR,
+            NimbusExperiment.Channel.AURORA,
+        )
+    )
+    def test_desktop_prefflips_128_esr_only(self, channel):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
+            application=NimbusExperiment.Application.DESKTOP,
+            firefox_min_version=NimbusExperiment.Version.FIREFOX_128,
+            firefox_max_version=NimbusExperiment.Version.NO_VERSION,
+            feature_configs=[
+                NimbusFeatureConfigFactory.create_desktop_prefflips_feature()
+            ],
+            channel=channel,
+        )
+
+        serializer = NimbusReviewSerializer(
+            experiment,
+            data=NimbusReviewSerializer(experiment, context={"user": self.user}).data,
+            context={"user", self.user},
+        )
+
+        if channel == NimbusExperiment.Channel.ESR:
+            self.assertTrue(serializer.is_valid(), serializer.errors)
+        else:
+            self.assertFalse(serializer.is_valid())
+            self.assertEqual(
+                serializer.errors,
+                {
+                    "firefox_min_version": [
+                        NimbusExperiment.ERROR_DESKTOP_PREFFLIPS_128_ESR_ONLY
+                    ],
+                },
+            )
+
+    @parameterized.expand(
+        (
+            NimbusExperiment.Channel.UNBRANDED,
+            NimbusExperiment.Channel.NIGHTLY,
+            NimbusExperiment.Channel.BETA,
+            NimbusExperiment.Channel.RELEASE,
+            NimbusExperiment.Channel.ESR,
+            NimbusExperiment.Channel.AURORA,
+        )
+    )
+    def test_desktop_prefflips_129(self, channel):
+        experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.CREATED,
+            targeting_config_slug=NimbusExperiment.TargetingConfig.NO_TARGETING,
+            application=NimbusExperiment.Application.DESKTOP,
+            firefox_min_version=NimbusExperiment.Version.FIREFOX_128,
```

Comment:
We should try to have the test names be almost like sentences describing waht they're testing.  Like this is testing 'test that desktop pref flips are valid for 129'?  It just says 'test desktop pref flips 129' but what about them? That they're valid? Invalid?  The test name here refers to 129, but the version in the test itself is 128, and then at the bottom asserts it's valid, for all channels, so I'm not sure what's expected here?  The ticket says it should only be valid for 128 in ESR, but that's not what this test is saying?

Reply:
Yeah that was supposed to be v129, which is why all the tests are failing :) bady copy+paste